### ISSUE TITLE
Why patching pybind11_bazel ?

### ISSUE THREAD

**Issue by Mizux** [Mizux/bazel-pybind11#4]
Try to explain why we have to patch pybind11_bazel.

TLDR: on Windows python native module must have `.pyd` as extension and not `.dll` which is not supported (ed hard coded) in bazel internal machinery.
fun fact: it is `.so` for macos...
ref: https://github.com/pypa/setuptools/blob/b545fc778583f644d6c331773dbe0ea53bfa41af/setuptools/_vendor/packaging/tags.py#L228-L241

ref: https://github.com/bazelbuild/bazel/blob/b9e1ec7431a20f33202303755c535819cce8ff1d/src/main/java/com/google/devtools/build/lib/rules/cpp/ArtifactCategory.java#L25 (missing .pyd here)

https://github.com/bazelbuild/bazel/blob/b9e1ec7431a20f33202303755c535819cce8ff1d/src/main/starlark/builtins_bzl/common/cc/cc_binary.bzl#L409 (missing .pyd here)

Few site fixed here. (7.0.0 preview...)
https://github.com/bazelbuild/bazel/commit/073f54b9a7c46afd2c28b4a99a235bdd6b63bb5f
